core/src/main/java/org/apache/gravitino/credential/CredentialCacheManager.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.credential;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.apache.gravitino.catalog.CatalogManager;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CredentialCacheManager<T> {
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogManager.class);
+
+  // Calculates the credential expire time in the cache.
+  static class CredentialExpireTimeCaculator<T> implements Expiry<T, Credential> {
+
+    private long credentialMaxCacheTimeInMs = 0;
+
+    public CredentialExpireTimeCaculator(long credentialMaxCacheTimeInMs) {
+      this.credentialMaxCacheTimeInMs = credentialMaxCacheTimeInMs;
+    }
+
+    // Set expire time after add a credential in the cache.
+    @Override
+    public long expireAfterCreate(
+        @NonNull T key, @NonNull Credential credential, long currentTime) {
```

Comment:
Curious about the `@NonNull` annotation ...
Can we use it extensively for validation of null params?

Reply:
removed